pyiron_base/database/filetable.py

```diff
@@ -56,8 +56,22 @@ class FileTableSingleton(ABCMeta):
     _instances = {}
 
     def __call__(cls, index_from):
-        _path = os.path.abspath(index_from)
+        _path = os.path.abspath(os.path.expanduser(index_from))
         if _path not in cls._instances:
+            path_match_lst = [
+                p for p in cls._instances.keys() if os.path.commonpath([_path, p]) == p
+            ]
+            if len(path_match_lst) > 0:
+                cls._instances[_path] = super(FileTableSingleton, cls).__call__(
```

Comment:
In fact, I think we can go even further and add a class method to avoid nested if-else in the main call function, so it looks like this:

```python
    def __call__(cls, index_from):
        path = os.path.abspath(os.path.expanduser(index_from))
        if path not in cls._instances:
            cls._instances[_path] = super(FileTableSingleton, cls).__call__(
                index_from=index_from,
                fileindex=cls._get_common_fileindex_if_available(path)
            )
        return cls._instances[path]
```

where 

```python
@classmethod
_get_common_fileindex_if_available(cls, path) -> PyFileIndex | None:
    common_path = cls._get_most_precise_common_path(
        path,
        cls._instances.keys() 
    )
    if common_path is None:
        return None
    else:
        return cls._instances[common_path]._fileindex.open(path)
```